lib/api/login.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    express = require('express');
+
+module.exports = loginRouterFactory;
+
+di.annotate(loginRouterFactory, new di.Provide('Http.Api.Login'));
+di.annotate(loginRouterFactory, new di.Inject(
+        'Auth.Services'
+    )
+);
+
+function loginRouterFactory (
+    authService
+) {
+
+
+    var router = express.Router();
+
+    /**
+     * @api {get} /login/ GET /
+     * @apiVersion 1.1.0
+     * @apiDescription get a token before accessing other apis
+     * @apiName login-get
+     * @apiGroup login
+     * @apiParam {query} username username used to login
+     * @apiParam {query} password password used to login
+     * @apiSuccess {json} a token to be used when accessing other apis:
+     *      {
+     *       "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoiY
+     *       WRtaW4iLCJpYXQiOjE0NTI0ODA3MTIsImV4cCI6MTQ1MjU2NzExMn0.Qo-9y
+     *       omlkd0WE1jlEA-L1WqA6k_9qJYhkicQL5IUmSg"
+     *      }
+     */
+
+    router.get('/login', authService.authMiddlewareLogin.bind(authService));
```

Comment:
Hi @brianparry , it's true that when you are sending the query using GET, a request body is not needed, instead, you have to provide the username and password in query string, as plain text, as you noticed. I guess you are concerning about security of having username/password in plain text but SSL by nature will secure the URL thus a man-in-the-middle attacker will not get to know the URL using a sniffer. Refer to [How secure are query string over https](https://blog.httpwatch.com/2009/02/20/how-secure-are-query-strings-over-https/). Please let me know if this does not address your concern. 

The reason we keep GET is that someone may use a plain web-browser (not using any browser plugins) to testify our APIs, in which case a POST body is not feasible. This seems to be a user case discussion thus I include @leoyjchang and @heckj for a comment here.